modules/navier_stokes/include/fvkernels/INSFVMomentumBoussinesq.h

```diff
@@ -37,5 +37,5 @@ class INSFVMomentumBoussinesq : public FVElementalKernel, public INSFVMomentumRe
   /// Reference temperature at which the value of _rho was measured
   const Real _ref_temperature;
   /// the density
-  const Real & _rho;
+  const Moose::Functor<ADReal> & _rho;
```

Comment:
I d rather we keep a few of these as reals. It will help make sure users never run INS with T,P dependent densities

Reply:
Added isConst() to the functors and a corresponding check/test for the Boussinesq term.